mds/apis/agency_api/v0_x/vehicles.py

```diff
@@ -93,7 +96,20 @@ class DeviceRegisterSerializer(serializers.Serializer):
     model = serializers.CharField(required=False, help_text="Vehicle Model.")
 
     def create(self, validated_data):
-        provider_id = self.context["request"].user.provider_id
+        data_broker_id = self.context["request"].user.provider_id
+
+        try:
+            provider_id = str(validated_data.pop("provider_id"))
+        except KeyError:
+            provider_id = data_broker_id
+
+        if data_broker_id != provider_id:
+            msg = "%s is trying to register a device with the provider id %s" % (
+                data_broker_id,
+                provider_id,
+            )
+            logger.warning(msg)
```

Comment:
Warnings will end up in Sentry, do we want them when there is nothing to fix? An "info" level seems enough.

Reply:
this is actually gonna be the **normal** code path. Do we even want to log it ?

Reply:
```suggestion
            logger.warning("%s is trying to register a device with the provider id %s", data_broker_id, provider_id)
```
le fait de ne pas interpoler directement les variables permet de ne créer que si il y a effectivement un log et permet à Sentry de détecter un pattern pour regrouper les erreurs